### PR TITLE
Fix empty documentation commits

### PR DESCRIPTION
**Pull request by andrii-suse**
https://progress.opensuse.org/issues/59957

First commit should fix cosmetic problem when too many documentation changes are detected in CI run for every PR

Before: https://circleci.com/gh/os-autoinst/openQA/6397 : `4 files changed, 13617 insertions(+), 9819 deletions(-)`
After: 'Documentation is up to date' if nothing changes, otherwise relevant diff info.

Second commit is challenged by inability of `git diff` to ignore specific lines and introduces more reliable hacks to detect actual changes.

### CODE REVIEW

File: script/generate-documentation [hunk 1]
```diff
@@ -46,6 +46,23 @@
...
+set -eo pipefail
+
+# exit status 0 if any changes detected
+anything_changed() {
+    git --no-pager diff --shortstat
+    
+    local lines_changed_in_index lines_changed_in_api other_files_changed
+
+    # it looks it is a challenge to ignore lines in git diff
+    lines_changed_in_index="$(git --no-pager diff -U0 -- index.html     | grep -v 'Last updated ' | grep -v -E '^\@\@' | grep -v -- '--- a' | grep -v '+++ b' | grep -v '^index ' | tail -n +2 | wc -l)"
```
Comment (Martchus) [inline script/generate-documentation]: I'm wondering where it is possible to combine these grep invocations into a single one.

File: script/generate-documentation [hunk 2]
```diff
@@ -46,6 +46,23 @@
 # target remote ($1). Target branch will be created based on
 # PULL_REQUEST_USER:gh_pages
 
+set -eo pipefail
+
+# exit status 0 if any changes detected
+anything_changed() {
+    git --no-pager diff --shortstat
+    
+    local lines_changed_in_index lines_changed_in_api other_files_changed
```
Comment (Martchus) [inline script/generate-documentation]: I personally prefer avoiding such lines and just use local on the first initialization.

**Comment by okurz** [comment]
It seems the scheduler test is unstable as well. https://github.com/os-autoinst/openQA/pull/2516 to do retries. I will retrigger the circle CI test here.

File: script/generate-documentation (hunk 2 above)
Comment (andrii-suse) [reply script/generate-documentation]: that I did as well, but shellcheck didn't like it for some reasons

Comment (Martchus) [reply script/generate-documentation]: Ok, then let's keep it although I'm wondering why shellcheck enforces this.

**Pull request merged**